File: javalin-micrometer/src/test/java/io/javalin/micrometer/MicrometerPluginTest.java
package io.javalin.micrometer;

import io.javalin.Javalin;
import io.micrometer.core.instrument.MeterRegistry;
import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MicrometerPluginTest {

    @Test
    public void testPluginCanBeRegistered() {
        Javalin app = Javalin.create(cfg -> {
            cfg.registerPlugin(new MicrometerPlugin());
        });
        app.get("/test", ctx -> ctx.result("Hello World"));
        
        // Just verify the plugin was registered without error
        assertThat(app).isNotNull();
        app.stop();
    }

    @Test
    public void testPluginWithCustomRegistry() {
        MeterRegistry registry = new SimpleMeterRegistry();
        
        Javalin app = Javalin.create(cfg -> {
            cfg.registerPlugin(new MicrometerPlugin(config -> {
                config.registry = registry;
                config.tags = new String[]{"service", "test-app"};
            }));
        });
        app.get("/test", ctx -> ctx.result("Hello World"));
        
        // Verify the registry was set correctly
        assertThat(registry).isNotNull();
        app.stop();
    }

    @Test
    public void testPluginWithDisabledFeatures() {
        Javalin app = Javalin.create(cfg -> {
            cfg.registerPlugin(new MicrometerPlugin(config -> {
                config.enableHttpMetrics = false;
            }));
        });
        app.get("/test", ctx -> ctx.result("Hello World"));
        
        // Just verify the plugin was configured without error
        assertThat(app).isNotNull();
        app.stop();
    }
}

File: javalin/src/test/kotlin/io/javalin/plugin/metrics/MicrometerPluginTest.kt
package io.javalin.plugin.metrics

import io.javalin.Javalin
import io.javalin.testing.TestUtil
import io.micrometer.core.instrument.simple.SimpleMeterRegistry
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.fail
import org.junit.Test

class MicrometerPluginTest {

    @Test
    fun `test that JettyConnectionMetrics is registered`() {
        val registry = SimpleMeterRegistry()

        TestUtil.test(Javalin.create { it.registerPlugin(MicrometerPlugin(registry)) }) { app, http ->
            app.get("/test") { ctx -> ctx.json("Hello world") }
            repeat(10) {
                http.get("/test")
            }
        }

        val maxConnectionGauge = registry.find("jetty.connections.max").gauge() ?: fail("\"jetty.connections.max\" not found")
        assertThat(maxConnectionGauge.value()).isGreaterThan(0.0)

        val messagesOutCounter = registry.find("jetty.connections.messages.out").counter() ?: fail("\"jetty.connections.messages.out\" not found")
        assertThat(messagesOutCounter.count()).isGreaterThan(0.0)
    }
}
